### ISSUE TITLE
Nothing happens when I try to upload a magnet link in transmission WebUI

### ISSUE THREAD

**Issue by cerealconyogurt** [haugene/docker-transmission-openvpn#1597]
**Describe the problem**

I was able to run the docker with the docker-file below, and I can access the WebUI without any problem. But when I add a magnet link and then I click on the upload button nothing happens. I was searching if someone had the same problem but I could not find any similar issue.

From the transmssion.log file, I can see a port forwarding issue. May it be the reason... any advice?
I can run another transmission docker without problems (ghcr.io/linuxserver/transmission), so it may be a problem with the setup I have for this container...

My local network is `192.168.1.0/24`
The NAS IP where I have the docker engine is `192.168.1.100` and the docker container IP is `172.21.0.2`

**My docker-compose file is:**

```
version: "3.7"
services:
  transmission:
    image: haugene/transmission-openvpn:latest
    container_name: transmission-vpn
    cap_add:
      - NET_ADMIN
    # devices:
    #  - /dev/net/tun
    dns:
      - 1.1.1.1
      - 1.0.0.1
    environment:
      PUID: $PUID
      PGID: $PGID
      TZ: $TZ
      UMASK_SET: 2
      LOCAL_NETWORK: "192.168.1.0/24"
      TRANSMISSION_WEB_UI: combustion # + kettu or transmission-web-control
      TRANSMISSION_RPC_AUTHENTICATION_REQUIRED: "true"
      TRANSMISSION_RPC_USERNAME: $TRANSMISSION_USER
      TRANSMISSION_RPC_PASSWORD: $TRANSMISSION_PASS
      TRANSMISSION_RPC_HOST_WHITELIST: "127.0.0.1,192.168.100.*"
      TRANSMISSION_UMASK: 002
      # TRANSMISSION_RATIO_LIMIT: 0.01
      TRANSMISSION_RATIO_LIMIT_ENABLED: "true"
      TRANSMISSION_ALT_SPEED_DOWN: 2000
      TRANSMISSION_ALT_SPEED_ENABLED: "true"
      TRANSMISSION_ALT_SPEED_UP: 15
      TRANSMISSION_SPEED_LIMIT_DOWN: 6000
      TRANSMISSION_SPEED_LIMIT_DOWN_ENABLED: "true"
      TRANSMISSION_SPEED_LIMIT_UP: 30
      TRANSMISSION_SPEED_LIMIT_UP_ENABLED: "true"
      TRANSMISSION_INCOMPLETE_DIR: /download/incomplete
      TRANSMISSION_INCOMPLETE_DIR_ENABLED: "true"
      TRANSMISSION_WATCH_DIR: /download/watch
      TRANSMISSION_WATCH_DIR_ENABLED: "true"
      TRANSMISSION_DOWNLOAD_DIR: /download/completed
      HEALTH_CHECK_HOST: www.google.com
      OPENVPN_PROVIDER: CUSTOM
      OPENVPN_USERNAME: $EXPRESSVPN_USER
      OPENVPN_PASSWORD: $EXPRESSVPN_PASS
      OPENVPN_CONFIG: my_expressvpn_argentina_udp
      OPENVPN_OPTS: --inactive 3600 --ping 10 --ping-exit 60
    volumes:
      - $DOCKERDIR/transmission_vpn/config:/config
      - $DOCKERDIR/transmission_vpn/custom:/etc/openvpn/custom
      - $DOCKERDIR/transmission_vpn/data:/data
      - $DOWNLOADDIR/transmission:/download
      - /etc/localtime:/etc/localtime:ro
    ports:
      - $IP_NAS:$TRANSMISSION_PORT:9091
    hostname: transmission
    restart: unless-stopped
```

**Logs**
```
Setting OpenVPN credentials...
adding route to local network 192.168.1.0/24 via 172.21.0.1 dev eth0
Tue Dec  8 13:06:00 2020 WARNING: --keysize is DEPRECATED and will be removed in OpenVPN 2.6
Tue Dec  8 13:06:00 2020 OpenVPN 2.4.9 x86_64-alpine-linux-musl [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Apr 20 2020
Tue Dec  8 13:06:00 2020 library versions: OpenSSL 1.1.1g  21 Apr 2020, LZO 2.10
Tue Dec  8 13:06:00 2020 WARNING: --ns-cert-type is DEPRECATED.  Use --remote-cert-tls instead.
Tue Dec  8 13:06:00 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Tue Dec  8 13:06:00 2020 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
Tue Dec  8 13:06:00 2020 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
Tue Dec  8 13:06:00 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]185.166.253.27:1195
Tue Dec  8 13:06:00 2020 Socket Buffers: R=[212992->425984] S=[212992->425984]
Tue Dec  8 13:06:00 2020 UDP link local: (not bound)
Tue Dec  8 13:06:00 2020 UDP link remote: [AF_INET]185.166.253.27:1195
Tue Dec  8 13:06:01 2020 TLS: Initial packet from [AF_INET]185.166.253.27:1195, sid=584bb874 b4d03486
Tue Dec  8 13:06:01 2020 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
Tue Dec  8 13:06:01 2020 VERIFY OK: depth=1, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=ExpressVPN CA, emailAddress=support@expressvpn.com
Tue Dec  8 13:06:01 2020 VERIFY OK: nsCertType=SERVER
Tue Dec  8 13:06:01 2020 VERIFY X509NAME OK: C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-7527-0a, emailAddress=support@expressvpn.com
Tue Dec  8 13:06:01 2020 VERIFY OK: depth=0, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-7527-0a, emailAddress=support@expressvpn.com
Tue Dec  8 13:06:01 2020 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 2048 bit RSA
Tue Dec  8 13:06:01 2020 [Server-7527-0a] Peer Connection Initiated with [AF_INET]185.166.253.27:1195
Tue Dec  8 13:06:02 2020 SENT CONTROL [Server-7527-0a]: 'PUSH_REQUEST' (status=1)
Tue Dec  8 13:06:02 2020 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 10.111.0.1,comp-lzo no,route 10.111.0.1,topology net30,ping 10,ping-restart 60,ifconfig 10.111.0.6 10.111.0.5,peer-id 0,cipher AES-256-GCM'
Tue Dec  8 13:06:02 2020 OPTIONS IMPORT: timers and/or timeouts modified
Tue Dec  8 13:06:02 2020 OPTIONS IMPORT: compression parms modified
Tue Dec  8 13:06:02 2020 OPTIONS IMPORT: --ifconfig/up options modified
Tue Dec  8 13:06:02 2020 OPTIONS IMPORT: route options modified
Tue Dec  8 13:06:02 2020 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Tue Dec  8 13:06:02 2020 OPTIONS IMPORT: peer-id set
Tue Dec  8 13:06:02 2020 OPTIONS IMPORT: adjusting link_mtu to 1629
Tue Dec  8 13:06:02 2020 OPTIONS IMPORT: data channel crypto options modified
Tue Dec  8 13:06:02 2020 Data Channel: using negotiated cipher 'AES-256-GCM'
Tue Dec  8 13:06:02 2020 NCP: overriding user-set keysize with default
Tue Dec  8 13:06:02 2020 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Tue Dec  8 13:06:02 2020 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Tue Dec  8 13:06:02 2020 ROUTE_GATEWAY 172.21.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:15:00:02
Tue Dec  8 13:06:02 2020 TUN/TAP device tun0 opened
Tue Dec  8 13:06:02 2020 TUN/TAP TX queue length set to 100
Tue Dec  8 13:06:02 2020 /sbin/ip link set dev tun0 up mtu 1500
Tue Dec  8 13:06:02 2020 /sbin/ip addr add dev tun0 local 10.111.0.6 peer 10.111.0.5
Tue Dec  8 13:06:02 2020 /etc/openvpn/tunnelUp.sh tun0 1500 1557 10.111.0.6 10.111.0.5 init
Up script executed with tun0 1500 1557 10.111.0.6 10.111.0.5 init
Updating TRANSMISSION_BIND_ADDRESS_IPV4 to the ip of tun0 : 10.111.0.6
Using Combustion UI, overriding TRANSMISSION_WEB_HOME
Updating Transmission settings.json with values from env variables
Using existing settings.json for Transmission /data/transmission-home/settings.json
Overriding alt-speed-down because TRANSMISSION_ALT_SPEED_DOWN is set to 2000
Overriding alt-speed-enabled because TRANSMISSION_ALT_SPEED_ENABLED is set to true
Overriding alt-speed-up because TRANSMISSION_ALT_SPEED_UP is set to 15
Overriding bind-address-ipv4 because TRANSMISSION_BIND_ADDRESS_IPV4 is set to 10.111.0.6
Overriding download-dir because TRANSMISSION_DOWNLOAD_DIR is set to /download/completed
Overriding incomplete-dir because TRANSMISSION_INCOMPLETE_DIR is set to /download/incomplete
Overriding incomplete-dir-enabled because TRANSMISSION_INCOMPLETE_DIR_ENABLED is set to true
Overriding ratio-limit-enabled because TRANSMISSION_RATIO_LIMIT_ENABLED is set to true
Overriding rpc-authentication-required because TRANSMISSION_RPC_AUTHENTICATION_REQUIRED is set to true
Overriding rpc-host-whitelist because TRANSMISSION_RPC_HOST_WHITELIST is set to 127.27.0.1,192.168.1.*,192.168.100.*
Overriding rpc-password because TRANSMISSION_RPC_PASSWORD is set to [REDACTED]
Overriding rpc-port because TRANSMISSION_RPC_PORT is set to 9091
Overriding rpc-username because TRANSMISSION_RPC_USERNAME is set to xxxxxx
Overriding speed-limit-down because TRANSMISSION_SPEED_LIMIT_DOWN is set to 6000
Overriding speed-limit-down-enabled because TRANSMISSION_SPEED_LIMIT_DOWN_ENABLED is set to true
Overriding speed-limit-up because TRANSMISSION_SPEED_LIMIT_UP is set to 30
Overriding speed-limit-up-enabled because TRANSMISSION_SPEED_LIMIT_UP_ENABLED is set to true
Overriding umask because TRANSMISSION_UMASK is set to 2
sed'ing True to true
Enforcing ownership on transmission config directories
Applying permissions to transmission config directories
Setting owner for transmission paths to 1000:1000
Setting permission for files (644) and directories (755)
Setting permission for watch directory (775) and its files (664)
-------------------------------------
Transmission will run as
-------------------------------------
User name:   abc
User uid:    1000
User gid:    1000
-------------------------------------
STARTING TRANSMISSION
Transmission startup script complete.
Tue Dec  8 13:06:05 2020 /sbin/ip route add 185.166.253.27/32 via 172.21.0.1
Tue Dec  8 13:06:05 2020 /sbin/ip route add 0.0.0.0/1 via 10.111.0.5
Tue Dec  8 13:06:05 2020 /sbin/ip route add 128.0.0.0/1 via 10.111.0.5
Tue Dec  8 13:06:05 2020 /sbin/ip route add 10.111.0.1/32 via 10.111.0.5
Tue Dec  8 13:06:05 2020 Initialization Sequence Completed
Tue Dec  8 14:06:00 2020 VERIFY OK: depth=1, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=ExpressVPN CA, emailAddress=support@expressvpn.com
Tue Dec  8 14:06:00 2020 VERIFY OK: nsCertType=SERVER
Tue Dec  8 14:06:00 2020 VERIFY X509NAME OK: C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-7527-0a, emailAddress=support@expressvpn.com
Tue Dec  8 14:06:00 2020 VERIFY OK: depth=0, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-7527-0a, emailAddress=support@expressvpn.com
Tue Dec  8 14:06:00 2020 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Tue Dec  8 14:06:00 2020 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Tue Dec  8 14:06:00 2020 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 2048 bit RSA
Tue Dec  8 15:06:00 2020 VERIFY OK: depth=1, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=ExpressVPN CA, emailAddress=support@expressvpn.com
Tue Dec  8 15:06:00 2020 VERIFY OK: nsCertType=SERVER
Tue Dec  8 15:06:00 2020 VERIFY X509NAME OK: C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-7527-0a, emailAddress=support@expressvpn.com
Tue Dec  8 15:06:00 2020 VERIFY OK: depth=0, C=VG, ST=BVI, O=ExpressVPN, OU=ExpressVPN, CN=Server-7527-0a, emailAddress=support@expressvpn.com
Tue Dec  8 15:06:00 2020 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Tue Dec  8 15:06:00 2020 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Tue Dec  8 15:06:00 2020 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 2048 bit RSA
```

This is the log from transmission.log:

```
[2020-12-08 13:05:52.747] Saved "/data/transmission-home/settings.json" (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/variant.c:1221)
[2020-12-08 13:05:52.748] DHT Not saving nodes, DHT not ready (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-dht.c:445)
[2020-12-08 13:05:52.748] Port Forwarding Stopped (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/port-forwarding.c:196)
[2020-12-08 13:06:05.199] Transmission 3.00 (bb6b5a062e) started (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/session.c:769)
[2020-12-08 13:06:05.199] RPC Server Adding address to whitelist: 127.27.0.1 (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:956)
[2020-12-08 13:06:05.200] RPC Server Adding address to whitelist: 192.168.1.* (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:956)
[2020-12-08 13:06:05.200] RPC Server Adding address to whitelist: 192.168.100.* (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:956)
[2020-12-08 13:06:05.200] RPC Server Adding address to whitelist: 127.0.0.1 (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:956)
[2020-12-08 13:06:05.200] RPC Server Adding address to whitelist: ::1 (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:956)
[2020-12-08 13:06:05.200] RPC Server Serving RPC and Web requests on 0.0.0.0:9091/transmission/ (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:1243)
[2020-12-08 13:06:05.200] RPC Server Whitelist enabled (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:1249)
[2020-12-08 13:06:05.200] RPC Server Password required (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/rpc-server.c:1254)
[2020-12-08 13:06:05.200] UDP Failed to set receive buffer: requested 4194304, got 425984 (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-udp.c:97)
[2020-12-08 13:06:05.200] UDP Please add the line "net.core.rmem_max = 4194304" to /etc/sysctl.conf (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-udp.c:99)
[2020-12-08 13:06:05.200] UDP Failed to set send buffer: requested 1048576, got 425984 (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-udp.c:105)
[2020-12-08 13:06:05.200] UDP Please add the line "net.core.wmem_max = 1048576" to /etc/sysctl.conf (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-udp.c:107)
[2020-12-08 13:06:05.200] DHT Reusing old id (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-dht.c:383)
[2020-12-08 13:06:05.200] DHT Bootstrapping from 107 IPv4 nodes (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/tr-dht.c:172)
[2020-12-08 13:06:05.200] Using settings from "/data/transmission-home" (/home/buildozer/aports/community/transmission/src/transmission-3.00/daemon/daemon.c:646)
[2020-12-08 13:06:05.200] Saved "/data/transmission-home/settings.json" (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/variant.c:1221)
[2020-12-08 13:06:05.200] transmission-daemon requiring authentication (/home/buildozer/aports/community/transmission/src/transmission-3.00/daemon/daemon.c:674)
[2020-12-08 13:06:05.200] Port Forwarding (NAT-PMP) initnatpmp succeeded (0) (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/natpmp.c:73)
[2020-12-08 13:06:05.200] Port Forwarding (NAT-PMP) sendpublicaddressrequest succeeded (2) (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/natpmp.c:73)
[2020-12-08 13:06:13.200] Port Forwarding State changed from "Not forwarded" to "Starting" (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/port-forwarding.c:106)
[2020-12-08 13:06:13.200] Port Forwarding State changed from "Starting" to "???" (/home/buildozer/aports/community/transmission/src/transmission-3.00/libtransmission/port-forwarding.c:106)
```

**Host system:**
Docker version 19.03.8 on Asustor NAS

**Comment by CtrlAltJesse** [comment]
Seem to be having the same issue here - where are you finding transmission.log?

**Comment by cerealconyogurt** [comment]
> Seem to be having the same issue here - where are you finding transmission.log?

Hi, you will find the file in `/data/transmission-home` folder

**Comment by haugene** [comment]
I'm not sure I'm able to help you to a solution here. I haven't seen this particular one before :thinking: 

What I always find interesting is if you can reproduce the error in a "clean" setup. Now that usually means starting it really vanilla, like described here: https://haugene.github.io/docker-transmission-openvpn/debug/

If you want to start a bit lighter you could just skip mounting your host volumes, that way Transmission would start with a clean state. There has been several issues lately that has been solved by clearing Transmission state. I can't see how that should be a bug in this image and I hope it's something in Transmission that will be fixed at some point.

Could you try reducing your docker-compose to this and see if it works then?
```
version: "3.7"
services:
  transmission:
    image: haugene/transmission-openvpn:latest
    container_name: transmission-vpn
    cap_add:
      - NET_ADMIN
    dns:
      - 1.1.1.1
      - 1.0.0.1
    environment:
      LOCAL_NETWORK: "192.168.1.0/24"
      OPENVPN_PROVIDER: CUSTOM
      OPENVPN_USERNAME: $EXPRESSVPN_USER
      OPENVPN_PASSWORD: $EXPRESSVPN_PASS
      OPENVPN_CONFIG: my_expressvpn_argentina_udp
    volumes:
      - $DOCKERDIR/transmission_vpn/custom:/etc/openvpn/custom
    ports:
      - $IP_NAS:$TRANSMISSION_PORT:9091
    hostname: transmission
    restart: unless-stopped
```

If that works then start adding back to your original config and see when it fails. And do a `docker-compose pull` before starting so you know you're on the latest version.

**Comment by cerealconyogurt** [comment]
Thank **haugene**. I will do as suggested. I will let you know the outcome

**Comment by cerealconyogurt** [comment]
It worked!, so I started to add more instruction, and then I could replicate the problem. The problem relied on the mounted volumes (the data volume and download volume). Thank you for your help!

**Comment by CtrlAltJesse** [comment]
Hi @cerealconyogurt - can you go into this a bit more, and am doing a bit more troubleshooting on the problem. It looks like I have exact issue based on your transmission.log.

Can you clarify the problem around mounted volumes? Was is simply some sort of permissions issues?

**Comment by CtrlAltJesse** [comment]
Nevermind, this was fixed by removing `TRANSMISSION_RATIO_LIMIT=2.00` in my docker-compose file.

As I noticed this in the logs: (possibly referenced issue in #1499)
`Traceback (most recent call last):
File "/etc/transmission/updateSettings.py", line 69, in <module>
env_value = setting_type(env_value)`
